ttnn/cpp/ttnn/operations/eltwise/quantization/quantization.cpp

```diff
@@ -30,17 +35,45 @@ Tensor QuantOp::invoke(
     // LLK quant kernel expects the reciprocal of the actual scale to avoid doing div on the device
     return ttnn::prim::binary_ng(
         queue_id,
-        input_a,
+        tt::tt_metal::is_block_float(a_dtype) ? ttnn::typecast(input_tensor, DataType::BFLOAT16) : input_tensor,
         1.0f / scale,
         binary_ng::BinaryOpType::QUANT,
-        output_dtype.value_or(DataType::INT32),
+        c_dtype,
         memory_config,
         optional_output_tensor,
         lhs_activations,
         rhs_activations,
         post_activations);
 }
 
+Tensor QuantOp::invoke(
+    QueueId queue_id,
+    const Tensor& input_tensor,
+    const Tensor& scale,
+    const int32_t zero_point,
+    const std::optional<int32_t> axis,
+    const std::optional<const DataType>& output_dtype,
+    const std::optional<MemoryConfig>& memory_config,
+    std::optional<Tensor> optional_output_tensor) {
+    const ttnn::DataType b_dtype = scale.get_dtype();
+
+    TT_FATAL(tt::tt_metal::is_floating_point(b_dtype), "Quantize only takes floating-point number scales");
+    TT_FATAL(!tt::tt_metal::is_block_float(b_dtype), "Unsupported scale tensor format");
+    TT_FATAL(scale.get_logical_volume() == 1u, "Per-tensor quantize only takes a single scale value");
+
+    // The alternative is to pass ttnn::reciprocal(scale) to binary_ng, but it's expensive
+    const float scale_scalar = [](const DataType b_dtype, const Tensor& scale) {
+        if (b_dtype == DataType::FLOAT32) {
+            return scale.to_vector<float>()[0];
+        } else {
+            return scale.to_vector<bfloat16>()[0].to_float();
+        }
+    }(b_dtype, scale);
```

Comment:
```suggestion
    const auto scale_scalar = b_dtype == DataType::FLOAT32 ?
        scale.to_vector<float>()[0] :
        scale.to_vector<bfloat16>()[0].to_float();
```
I assume you've probably seen this immediately-invoked lambda pattern somewhere else, but it's only useful if you need `if constexpr` so the return type can be dependent and ill-formed instantiations can be discarded.

Reply:
Done